### PR TITLE
Add ``DeprecationWarning`` contextmanager to tests of deprecated methods

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Description

Removes some warnings from the `pytest` output that were bugging me.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Shouldn't we remove the use of the deprecated function inside ``is_sys_guard`` instead ?

**Comment by DanielNoord** [comment]
> Shouldn't we remove the use of the deprecated function inside `is_sys_guard` instead ?

As long as the functions are still in our codebase the tests are still relevant. We don't use the functions ourselves, we only test whether they do what they are supposed to be doing even if they are deprecated.

**Comment by Pierre-Sassoulas** [comment]
> As long as the functions are still in our codebase the tests are still relevant

Ho my bad I thought it was pylint.

**Review by Pierre-Sassoulas — APPROVED** [review]

File: tests/unittest_nodes.py [hunk 1]
```diff
@@ -386,12 +386,19 @@ def test_if_sys_guard() -> None:
         assert isinstance(code, list) and len(code) == 3
 
         assert isinstance(code[0], nodes.If)
-        assert code[0].is_sys_guard() is True
+        with pytest.warns(DeprecationWarning) as records:
+            assert code[0].is_sys_guard() is True
+            assert len(records) == 1
```
Comment (cdce8p) [inline tests/unittest_nodes.py:391]: These to methods will be removed soon, so no real need to assert each Deprecation warning. Why not use a filter decorator on the test functions?

```py
    @staticmethod
    @pytest.mark.filterwarnings("ignore:.*is_sys_guard:DeprecationWarning")
    def test_if_sys_guard() -> None:
        ...

    @staticmethod
    @pytest.mark.filterwarnings("ignore:.*is_typing_guard:DeprecationWarning")
    def test_if_typing_guard() -> None:
        ...
```

**Review by cdce8p — CHANGES_REQUESTED** [review]

**Review by cdce8p — APPROVED** [review]

**Pull request merged**